R/db.R

```diff
@@ -297,6 +297,14 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns) {
   }
 
   write_column_hashed_data <- function(df, tblname, append = TRUE) {
+    ## Don't cache anything that is in the blacklist.
+    if (!is.null(blacklist)) {
+      if (any(is.na(blacklist))) {
+        df <- df[apply(df, 1, function(x) all(!(is.na(x)))),]
+      }
+      df <- df[apply(df, 1, function(x) all(!(x %in% blacklist))),]
```

Comment:
And yes, the `|` operator is overloaded to work on matrices.

```r
> Reduce(`|`, lapply(blacklist, `==`, df))
        a     b     c
[1,] TRUE  TRUE    NA
[2,] TRUE  TRUE FALSE
[3,]   NA FALSE FALSE
```

Note `apply` was OK to use in my version above because it is on a legitimate *matrix* not a data.frame, so we do not occur the large speed penalty of type coercion.